checks/process_test.go

```diff
@@ -25,6 +28,123 @@ func makeProcess(pid int32, cmdline string) *process.FilledProcess {
 	}
 }
 
+func TestProcessMessages(t *testing.T) {
+	p := []*process.FilledProcess{
+		makeProcess(1, "git clone google.com"),
+		makeProcess(2, "mine-bitcoins -all -x"),
+		makeProcess(3, "foo --version"),
+		makeProcess(4, "foo -bar -bim"),
+		makeProcess(5, "datadog-process-agent -ddconfig datadog.conf"),
+	}
+	c := []*containers.Container{
+		makeContainer("foo"),
+		makeContainer("bar"),
+	}
+	// first container runs pid1 and pid2
+	c[0].Pids = []int32{1, 2}
+	c[1].Pids = []int32{3}
+	lastRun := time.Now().Add(-5 * time.Second)
+	syst1, syst2 := cpu.TimesStat{}, cpu.TimesStat{}
+	cfg := config.NewDefaultAgentConfig()
+	sysInfo := &model.SystemInfo{}
+	lastCtrRates := util.ExtractContainerRateMetric(c)
+
+	for i, tc := range []struct {
+		testName        string
+		cur, last       map[int32]*process.FilledProcess
+		containers      []*containers.Container
+		maxSize         int
+		blacklist       []string
+		expectedChunks  int
+		totalProcs      int
+		totalContainers int
+	}{
+		{
+			testName:        "no containers",
+			cur:             map[int32]*process.FilledProcess{p[0].Pid: p[0], p[1].Pid: p[1], p[2].Pid: p[2]},
+			last:            map[int32]*process.FilledProcess{p[0].Pid: p[0], p[1].Pid: p[1], p[2].Pid: p[2]},
+			maxSize:         2,
+			containers:      []*containers.Container{},
+			blacklist:       []string{},
+			expectedChunks:  2,
```

Comment:
I think we should rewrite these tests to actually confirm that the processes are grouped with the correct containers, as expected, to increase confidence in this PR, instead of just using hardcoded pre-calculated result values.

It might also be useful to have some generation helper functions like to `generateProcesses(100)` to test more real world scenarios where `maxSize` is 100.

Reply:
@sunhay A test is added to address this.